go.mod

```diff
@@ -144,9 +147,15 @@ replace (
 	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.31.2
 	k8s.io/kubectl => k8s.io/kubectl v0.31.2
 	k8s.io/kubelet => k8s.io/kubelet v0.31.2
+	k8s.io/kubernetes => k8s.io/kubernetes v1.31.2
 	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.31.2
 	k8s.io/metrics => k8s.io/metrics v0.31.2
 	k8s.io/mount-utils => k8s.io/mount-utils v0.31.2
 	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.31.2
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.31.2
 )
+
+replace (
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20241107164952-923091dd2b1a // release-4.18
+	sigs.k8s.io/structured-merge-diff/v4 => sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // remove when bumping k8s.io/apimachinery => v0.32.0
```

Comment:
why is this needed?

Reply:
keep API compatibility, we have other dependencies which cannot use the latest version